### DISCUSSION TITLE
WSLg graphical apps don't start if 'export DISPLAY=' in .bashrc

### DISCUSSION THREAD

**Discussion by n1cholson** [microsoft/wslg#282] [General]
This might help some of you get WSLg to work.  
In WSL2, prior to WSLg it was necessary to add an export DISPLAY= entry to your .bashrc if you wanted to use VcXsrv.

Those recommended entries look like this:
  export DISPLAY=$(/sbin/ip route | awk '/default/ { print $3 }'):0
  OR
  export DISPLAY="$(grep nameserver /etc/resolv.conf | sed 's/nameserver //'):0"
and you'll find many sites recommend one of these for WSL2 with VcXsrv.

With WSLg I found that removing these lines allowed X apps (gedit or xeyes for example) to work perfectly.
I started by just renaming .bashrc to .bashrc_old and then starting an Ubuntu command prompt.  It worked fine so I did a bisect search to figure out what was the offending line.

My suggestion would be to add something to the WSLg instructions to tell users to remove any prior "export DISPLAY" command. Better still would be to prompt the user whether it's OK to comment those lines out during installation.

**Comment by onomatopellan** [comment]
Take a look at https://github.com/microsoft/wslg/wiki
It's being updated frequently with that type of info.

**Reply by n1cholson** [reply]
Didn't find anything helpful there.

**Reply by onomatopellan** [reply]
The explanation is [this.](https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg#display-environment-variable)
You only need `export DISPLAY=:0`. You can run it inside the terminal before running a GUI app or just add it as last line in the `.bashrc` file.

**Reply by n1cholson** [reply]
I've been able to run X applications without that line.  
The problem I'm having now is that OpenGL doesn't work at all in WSLg


$:~$ echo $DISPLAY
:0
$:~$ glxgears
Error: glXCreateContext failed
$:~$

No OpenGL support is a deal breaker for WSLg
